pkg/k8s/watchers/endpointsynchronizer.go

```diff
@@ -371,9 +370,13 @@ func updateCEPUID(scopedLog *logrus.Entry, e *endpoint.Endpoint, localCEP *ciliu
 	if podHostIP == "" {
 		return fmt.Errorf("endpoint sync cannot take ownership of CEP: no pod HostIP")
 	}
-	if podHostIP != nodeIP {
-		return fmt.Errorf("endpoint sync cannot take ownership of CEP that is not local: CEP's pod %q, pod's hostIP %q, cilium nodeIP %q)",
-			e.GetK8sPodName(), podHostIP, nodeIP)
+	if nodeIP := node.GetCiliumEndpointNodeIP(); podHostIP != nodeIP {
```

Comment:
Can we use `GetIPv4().String()` instead of `GetCiliumEndpointNodeIP()` to make it more clear what's happening here (if we're going to be checking both ip types)?